t/27-consoles-s3270.t

```diff
@@ -77,8 +77,48 @@ subtest 's3270_console connect_and_login' => sub {
         [warnings { $s3270_console->connect_and_login(); }],
         bag(
             re(qr/connect_and_login.*\nRECONNECT.*\n/), re(qr/trying hard shutdown and reconnect.*/),
-        ), 'got reboot attempt',
+        ), 'reconnect attempt',
     );
+
+    $s3270_console_mock->redefine(expect_3270 => sub ($self, %arg) {
+            return ['Fill in your USERID and PASSWORD and press ENTER'] unless keys %arg == 1 && exists $arg{buffer_ready};
+            return ['RECONNECT'];
+    });
+    cmp_deeply(
+        [
+            warnings {
+                throws_ok {
+                    $s3270_console->connect_and_login();
+                } qr/Could not reclaim.*\n.*/, 'Dies';
+            }
+        ],
+        bag(
+            re(qr/connect_and_login.*\nRECONNECT.*\n/),
+            re(qr/trying hard shutdown and reconnect.*/),
+            re(qr/trying hard shutdown and reconnect.*/),
+            re(qr/connect_and_login.*\nRECONNECT.*\n/),
+            re(qr/connect_and_login.*\nRECONNECT.*\n/),
+            re(qr/Still connected, it's s390, so.*/),
```

Comment:
@perlpunk Any suggestion to make it even better?
The messages are repeated, that's why I have to explicitly write them the number of times they appear on stderr.

Reply:
I guess it's ok to be more explicit and repetitive in tests

Reply:
You could get at least rid of some superfluous stuff at the end of the regexes:
```suggestion
            re(qr/connect_and_login.*\nRECONNECT/),
            re(qr/trying hard shutdown and reconnect/),
            re(qr/trying hard shutdown and reconnect/),
            re(qr/connect_and_login.*\nRECONNECT/),
            re(qr/connect_and_login.*\nRECONNECT/),
            re(qr/Still connected, it's s390, so/),
```